### DISCUSSION TITLE
Config Context issue with credentials per device basis

### DISCUSSION THREAD

**Discussion by calvinrobson** [netbox-community/netbox#5633] [Q&A] [answered]
Hello,

I have recently began using netbox and I have searched everywhere for this.
I am having an issue with credentials on a per devices basis, I have tried to place a username and password into the "Config Context" tab but it is not working.

I have used the JSON format:

{
            "NAPALM_USERNAME": "username",
            "NAPALM_PASSWORD": "password"
}

it is stating that I have used "incorrect authentication" can you please help me with what I am possibly doing wrong?

Is there a different format, if so, how should it be done?

Thanks,

**Comment by cpmills1975** [comment, marked as answer]
AFAIK there is no support for per-device NAPALM settings (but have a feeling it has been raised as an FR). The config context is just pure JSON designed to be consumed by automation tools etc - Ansible for example. It has no bearing on the NAPALM configuration used by NetBox to talk to devices.

**Comment by candlerb** [comment]
Some discussion here:
https://groups.google.com/g/netbox-discuss/c/qh1_-BlIJw4/m/xGSK1cMrDQAJ
